pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -740,6 +744,24 @@ func (aws *AWSCloud) InstanceID(name string) (string, error) {
 	}
 }
 
+// InstanceType returns the type of the specified instance.
+func (aws *AWSCloud) InstanceType(name string) (string, error) {
+	awsInstance, err := aws.getSelfAWSInstance()
```

Comment:
I'm following the pattern from other functions, such as InstanceID(). They all let you call them from either the node itself or from another (e.g. a controller). metadata.GetMetadata() is local only.